treeherder/webapp/api/jobs.py

```diff
@@ -91,10 +95,20 @@ def list(self, request, project, jm):
         - return_type (dict)
         """
         MAX_JOBS_COUNT = 2000
+        EARLIEST_LAST_MODIFIED = datetime.datetime.utcnow() - \
+                                 datetime.timedelta(minutes=30)
+
         filter = UrlQueryFilter(request.query_params)
 
         offset = int(filter.pop("offset", 0))
         count = int(filter.pop("count", 10))
+        if "last_modified" in filter.conditions:
+            datestr = filter.get("last_modified")[1]
+            last_modified = parser.parse(datestr)
+            if last_modified < EARLIEST_LAST_MODIFIED:
+                return Response(
+                    "last_modified of {} is not within last 30 minutes".format(datestr),
```

Comment:
Should probably use the timedelta value here, instead of hardcoding 30 minutes.

E.g. "last_modified of {} is not within time window of {}".format(datestr, EARLIEST_LAST_MODIFIED)